sqlgen/matcher.go

```diff
@@ -0,0 +1,138 @@
+package sqlgen
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/samsarahq/thunder/internal"
+)
+
+// extractColumns extracts the columns used in a filter.
+func extractColumns(filter Filter) []string {
+	columns := make([]string, 0, len(filter))
+	for column := range filter {
+		columns = append(columns, column)
+	}
+	sort.Strings(columns)
```

Comment:
necessary, to make sure that we group queries with same sets of columns (as otherwise column order might differ because of go's random map key iteration order). added comment.